src/semisync_master.h

```diff
@@ -0,0 +1,115 @@
+#pragma once
+
+#include <thread>
+#include <vector>
+#include <memory>
+#include <string>
+#include <mutex>
+#include <atomic>
+#include <condition_variable>
+#include <set>
+#include <list>
+#include <event2/bufferevent.h>
+
+#include "replication.h"
+#include "config.h"
+
+struct WaitingNode {
+  uint64_t log_pos;
+  std::condition_variable cond;
+  int waiters;
+};
+
+struct WaitingNodeComparator {
+  bool operator()(struct WaitingNode* left, struct WaitingNode* right) {
```

Comment:
```suggestion
  bool operator()(struct WaitingNode* left, struct WaitingNode* right) const {
```

fix warning (the specified comparator type does not provide a const call operator)